src/averageAges.js

```diff
@@ -55,7 +75,28 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const neededPersons = onlyWithSon
+    ? people.filter(person => person.sex === 'm')
+    : people;
+
+  const ages = neededPersons.map((person) => {
+    const mother = person.mother;
+
+    const findMother = female => {
+      return mother === female.name;
+    };
+
+    const differenceAges = people.find(findMother)
+      ? person.born - people.find(findMother).born
+      : undefined;
+
+    return differenceAges;
+  });
+
+  const filterAges = ages.filter(age => age !== undefined);
+
+  return filterAges.reduce((sum, currentNum) =>
+    sum + currentNum, 0) / filterAges.length;
```

Comment:
maybe you can use helper func in this case?

Reply:
now I can't use this function, because function calculates the age of each person, and in this caseI need to calculate the difference between mother and child (son)